libs/community/langchain_community/chat_models/litellm.py

```diff
@@ -410,6 +415,32 @@ async def _agenerate(
         )
         return self._create_chat_result(response)
 
+    def bind_tools(
+        self,
+        tools: Sequence[Union[Dict[str, Any], Type[BaseModel], Callable, BaseTool]],
+        **kwargs: Any,
+    ) -> Runnable[LanguageModelInput, BaseMessage]:
+        """Bind tool-like objects to this chat model.
+
+        LiteLLM expects tools argument in OpenAI format.
+
+        Args:
+            tools: A list of tool definitions to bind to this chat model.
+                Can be  a dictionary, pydantic model, callable, or BaseTool. Pydantic
+                models, callables, and BaseTools will be automatically converted to
+                their schema dictionary representation.
+            tool_choice: Which tool to require the model to call.
```

Comment:
Was `tool_choice` meant to be implemented?

Reply:
@ccurme thanks for having a look!

The changes are inspired by this example: https://github.com/langchain-ai/langchain/blob/master/libs/partners/mistralai/langchain_mistralai/chat_models.py#L617

`tools` and `tool_choice` are mentioned as accepted params. The `tools` param is formatted while `tool_choice` is passed as-is. According to https://litellm.vercel.app/docs/completion/function_call `tool_choice` can be passed as-is and doesn't need additional processing like here: https://github.com/langchain-ai/langchain/blob/0e916d0d552fde282fa3bf51010c47536913bba3/libs/partners/anthropic/langchain_anthropic/chat_models.py#L875